client/src/services/notificationCenter.ts

```diff
@@ -10,84 +14,79 @@ export enum NotificationLevel {
   Error = 'ERROR',
 }
 
-export interface Notification {
-  id: string,
-  title?: string,
-  message: string,
-  level: NotificationLevel,
-  read: boolean,
-  time: DateTime,
-  data: object,
+export class Notification {
+  id: string;
+  time: DateTime;
+  read: boolean;
+  constructor(public message: string,
+    public level: NotificationLevel,
+    public trace: boolean,
+    public data: object,
+    public title?: string,) {
+      this.id = uuid4()
+      this.time = DateTime.now()
+      this.read = false;
+    }
 }
 
 export class NotificationCenter {
   notifications: Notification[];
 
-  constructor() {
+  constructor(public t: ComposerTranslation) {
     this.notifications = [];
+    this.t = t;
   }
 
-  async showModal(
-    title: string,
-    message: string,
-    level: NotificationLevel,
-    addToList: boolean,
-    trace = false,
-    data = {},
-  ): Promise<void> {
-    const notif: Notification = {
-      id: uuid4(),
-      title: title,
-      message: message,
-      level: level,
-      read: false,
-      time: DateTime.now(),
-      data: data,
-    };
-    if (trace) {
+  async showModal(notif: Notification): Promise<void> {
+    if (notif.trace) {
       this.trace(notif);
     }
-    if (addToList) {
-      this.notifications.push(notif);
+    const modalConfig: MsModalConfig = {
+      theme: this.getModalTheme(notif.level),
+      subtitle: notif.message,
+      closeButtonEnabled: true,
+      confirmButton: {
+        disabled: false,
+        label: this.t('Notification.nextButton'),
+        onClick: () => {
+          this.markAsRead(notif.id);
+          modalController.dismiss();
+        },
+      },
+    };
+    if (notif.title) {
+      modalConfig.title = notif.title;
     }
+    const modal = await modalController.create({
+      component: MsModal,
+      canDismiss: true,
+      cssClass: 'notification-modal',
+      componentProps: modalConfig,
+    });
+    await modal.present();
+    const result = await modal.onWillDismiss();
   }
 
-  async showSnackbar(
-    message: string,
-    level: NotificationLevel,
-    addToList: boolean,
-    trace = false,
-    data = {},
-  ): Promise<void> {
-    const notif: Notification = {
-      id: uuid4(),
-      message: message,
-      level: level,
-      read: false,
-      time: DateTime.now(),
-      data: data,
-    };
-    if (trace) {
+  async showSnackbar(notif: Notification): Promise<void> {
+    if (notif.trace) {
       this.trace(notif);
     }
-    if (addToList) {
-      this.notifications.push(notif);
-    }
   }
 
-  async addToList(title: string, message: string, level: NotificationLevel, trace = false, data = {}): Promise<void> {
-    const notif: Notification = {
-      id: uuid4(),
-      title: title,
-      message: message,
-      level: level,
-      read: false,
-      time: DateTime.now(),
-      data: data,
-    };
-    if (trace) {
-      this.trace(notif);
+  getModalTheme(notificationLevel: NotificationLevel): MsModalTheme {
```

Comment:
Is there a way to make methods private or hidden? I usually just put an `_` at the start of its name like in Python. It's just to have an indication that you shouldn't call this method outside of this class (or at least that there's no point in doing so).

Reply:
Yes, by writing private before the function name. the '_' prefix is also commin in JS